### PR TITLE
[SPARK-47166][INFRA] Improves the HINTs of merge_spark_pr.py

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This improves the UX of merge_spark_pr.py

- Print errors in red in order to get attention effectively 
image
- Print prompt in bold
- Print the default option for `y/n` from prompt in upper case

### Why are the changes needed?

better dev UX

### Does this PR introduce _any_ user-facing change?

no, committer only

### How was this patch tested?

manually

image

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

File: dev/merge_spark_pr.py [hunk 1]
```diff
@@ -72,6 +72,13 @@
 BRANCH_PREFIX = "PR_TOOL"
 
 
+def print_error(msg):
+    print("\033[91m%s\033[0m" % msg)
+
+
+def bold_input(prompt) -> str:
+    return input("\033[1m%s\033[0m" % prompt)
+
```
Comment (dongjoon-hyun) [inline dev/merge_spark_pr.py:81]: nit. We need two empty new lines, don't we?

File: dev/merge_spark_pr.py [hunk 2]
```diff
@@ -72,6 +72,13 @@
 BRANCH_PREFIX = "PR_TOOL"
 
 
+def print_error(msg):
+    print("\033[91m%s\033[0m" % msg)
```
Comment (dongjoon-hyun) [inline dev/merge_spark_pr.py:76]: Just a question. Does this color code work in all shell and OS environments?

File: dev/merge_spark_pr.py [hunk 3]
```diff
@@ -604,7 +611,7 @@ def main():
         print("I've re-written the title as follows to match the standard format:")
         print("Original: %s" % pr["title"])
         print("Modified: %s" % modified_title)
-        result = input("Would you like to use the modified title? (y/n): ")
+        result = bold_input("Would you like to use the modified title? (y/N): ")
```
Comment (dongjoon-hyun) [inline dev/merge_spark_pr.py:614]: The default value suggestion looks nice to me.

**Review by dongjoon-hyun — COMMENTED** [review]
Please fix the Python linter failure, @yaooqinn .

File: dev/merge_spark_pr.py (hunk 2 above)
Comment (yaooqinn) [reply dev/merge_spark_pr.py:76]: https://en.wikipedia.org/wiki/ANSI_escape_code#Colors

Not all terminals and operating systems, but most of them support ANSI colors

File: dev/merge_spark_pr.py (hunk 1 above)
Comment (yaooqinn) [reply dev/merge_spark_pr.py:81]: fixed

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Thank you, @yaooqinn .

**Comment by yaooqinn** [comment]
Thank you @dongjoon-hyun, merged to master

**Pull request closed without merging**